doc/tutorials/sgx_ virtualization.rst

```diff
@@ -0,0 +1,213 @@
+software access, including privileged malware.
+
+
+High Level ACRN SGX Virtualization Design
+*****************************************
+
+ACRN SGX virtualization support can be divided into three parts: 
+
+* SGX capability exposed to Guest EPC (Enclave Page Cache) management Enclave
+* System function handing
+
+The image below shows the high-level design of SGX virtualization in ACRN. 
+
+.. figure:: images/sgx-1.png
+   :width: 500px
+   :align: left
+
+   Figure 1: SGX Virtualization in ACRN
+
+
+Enable SGX support for Guest 
+****************************
+
+Presumptions 
+============
+
+No Enclave in a Hypervisor 
+--------------------------
+
+ACRN does not support running an enclave in a hypervisor since currently the
+whole hypervisor is running in vmx root mode, ring 0, and an enclave must run
+in ring 3. SGX virtualization in ACRN provides the SGX capability to (non-SOS)
+VMs.
+
+Enable SGX on Host 
+------------------
+
+In order to enable SGX virtualization support in ACRN, the SGX feature must be
+enabled and the Processor Reserved Memory (PRM) must be configured in the
+platform BIOS. Note that ACRN does not support the "Software Control" option;
+SGX must be enabled manually in the BIOS. If the hardware platform does not
+support SGX or if SGX is not enabled in the BIOS, the SGX virtualization is
+not enabled.
+
+EPC Page Swapping in Guest 
+--------------------------
+
+ACRN only partitions the physical EPC resources for VMs. The Guest OS kernel
+handles EPC Page Swapping inside Guest.
+
+Instructions 
+============
+
+By default, SGX support for Guest is not enabled on the Master branch. Thus,
+you must perform the following:
+
+#. Make sure SGX is supported on the platform (CPUID.07H.EBX[2] should be 1).
+#. Opt-in to the SGX feature in the BIOS. Follow these instructions: 
+
+   a) Go to the Security page:
+
+      .. figure:: images/sgx-2.jpg
+         :width: 500px
+         :align: left
+
+   #) Enable SGX and configure the SGX Reserved Memory size as below: 
+
+      * Intel Software Guard Extension (SGX) -> Enabled 
+      * SGX Reserved Memory Size -> 128MB
+
+      .. figure:: images/sgx-3.jpg
+         :width: 500px
+         :align: left
+
+      .. note:: 
+         Not all SGX Reserved Memory can be used as EPC. On KBL-NUC-i7,
+         the SGX EPC size is 0x5d80000 (93.5MB) when the SGX Reserved Memory Size is set to 128MB. 
+
+#. Add the EPC config in the VM configuration:
+   Apply the patch to enable SGX support in UOS in the SDC scenario:
+   https://github.com/binbinwu1/acrn-hypervisor/commit/0153b2b9b9920b61780163f19c6f5318562215ef.patch
+
+#. Enable SGX in Guest:
+
+   * Refer to https://github.com/intel/linux-sgx on how to enable SGX in Linux Guest.
+   * Refer to https://software.intel.com/en-us/articles/getting-started-with-sgx-sdk-for-windows on how to enable SGX in Windows Guest.
+
+SGX Capability Exposure
+***********************
+
```

Comment:
Needs some introductory material here explaining what's being documented in the following sections

Reply:
ACRN exposes SGX capability and EPC resource to guest via MSRs and CPUIDs.